volatility/framework/interfaces/objects.py

```diff
@@ -176,6 +176,32 @@ def has_member(self, member_name: str) -> bool:
         """
         return False
 
+    def has_valid_member(self, member_name: str) -> bool:
+        """Returns whether the dereferenced type has a valid member."""
+        if self.has_member(member_name):
+            try:
+                _ = getattr(self, member_name)
+                return True
+            finally:
+                return False
+        return False
+
+    def has_valid_members(self, member_names: List[str]) -> bool:
+        """Returns whether the object has all of the members listed in member_names
+
+        Args:
+            member_names: List of names to test as to members with those names validity
+        """
+        # Use only a single try/except for efficiency
+        try:
+            for member_name in member_names:
+                if not self.has_member(member_name):
```

Comment:
can we just call has_valid_member for each element in the loop vs repeat code?

Reply:
The reason for not doing it was to avoid constructing loads of try/except blocks, rather than just one.  It can actually stack up if this is going to be called often, which it will, so for efficiency reasons I'm going to leave it that way.  There's not a lot of code duplicated and the speed savings could end up being significant...

Reply:
Hmmm, having moved to a context manager, and the fact that these methods may be reimplemented by higher up classes, we'll give the looped call a go and see how it works out...  5:)

Reply:
Ok, may have to give that a rethink.  Setting up the try/catch is minimal, but catching an exception is 8x slower than just letting it through.  Turns out, the contextlib is always at least 25x slower, so I'm gonna go back to the exception route, "too broad" exceptions be damned...  5:P

Reply:
For reference, adding a try/except clause (that doesn't fire) adds about 6 nanoseconds on my i7-8656U system.  So millions of those probably still won't add up to a very significant slowdown.